test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  MetadataAgentConfiguration config;
+  static void ParseConfiguration(MetadataAgentConfiguration& config, const std::string configString) {
+    std::stringstream stream(configString);
+    config.ParseConfiguration(stream);
+  }
+};
+
+static void TestDefaultConfig(MetadataAgentConfiguration& config) {
```

Comment:
Also didn't notice earlier: this can take a `const` reference. However, if you make it an instance function, you should instead make it a `const` function (i.e.,
```cpp
void VerifyDefaultConfig() const {
  ...
}
```
).

Reply:
Done.